content/en/blog/2021/stateful-set-made-easier/index.md

```diff
@@ -0,0 +1,223 @@
+---
+title: "StatefulSets Made Easier With Istio 1.10"
+description: Learn how to easily deploy stateful sets with Istio 1.10.
+publishdate: 2021-05-18
+attribution: "Lin Sun (Solo.io), Christian Posta (Solo.io), John Howard (Google), Zhonghu Xu (Huawei)"
+keywords: [statefulset,Istio,networking,localhost,loopback,eth0]
+---
+
+Kubernetes [StatefulSets](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) are commonly used to manage stateful applications. In addition to managing the deployment and scaling of a set of Pods, StatefulSets provide guarantees about the ordering and uniqueness of these Pods. Common StatefulSets used with Istio are ZooKeeper, Cassandra, Elasticsearch, Redis, Apache NiFi, and more. The Istio community has been making gradual progress towards supporting zero-configuration for StatefulSets, from automatic mTLS to eliminating any ServiceEntry or Istio resources, to the most recent [pod networking change in Istio 1.10](/blog/2021/upcoming-networking-changes/).
+
+## Challenges With StatefulSet
+
+Many [issues](https://github.com/istio/istio/issues/10659) have been reported from the Istio community around StatefulSets. What is unique about StatefulSet and service mesh? The StatefulSet pods are created from the same spec, but are not interchangeable: each has a persistent identifier that it maintains across any rescheduling. Pods within a StatefulSet listen on the pod IP only instead of `0.0.0.0` for communication within the StatefulSet pods.
```

Comment:
This is true for stateful sets in general.  also mentioned in @howardjohn 's blog. I think it is related to how stateful sets pods are created from the same spec and didn't want to listen on loopback interface for that inter-pod communication.

Reply:
This is not accurate, any application, statefulset or otherwise, can bind to any interface it chooses. It just happens to be common among applications that happen to use statefulsets

Reply:
@craigbox @howardjohn anything you propose to change here?  I didn't cover all pods/apps as John covered in his other blog.  I think what we have around stateful sets is accurate in this blog.

Reply:
`Pods within a StatefulSet listen on the pod IP only instead of `0.0.0.0` for communication within the StatefulSet pods.` is not accurate. It would be accurate to say `It is common for pods within a StatefulSet listen on the pod IP only, instead of 0.0.0.0, for communication within the StatefulSet pods`, but it certainly doesn't apply to all statefulsets.

Statefulsets are actually just normal pods with a few fields set BTW, you can do everything with a statefulset with Pod

Reply:
Yah, I like that, thanks for that clarification! you'll never know how users can configure a StatefulSet :)